Java/jni/native_fleece.cc

```diff
@@ -339,7 +340,20 @@ Java_com_couchbase_litecore_fleece_FLValue_asDouble(JNIEnv *env, jclass clazz, j
 JNIEXPORT jstring JNICALL
 Java_com_couchbase_litecore_fleece_FLValue_asString(JNIEnv *env, jclass clazz, jlong jvalue) {
     FLString str = FLValue_AsString((FLValue) jvalue);
-    return toJString(env, {str.buf, str.size});
+    const char* newBytes;
+    size_t len = UTF8ToModifiedUTF8((const char*)str.buf, &newBytes, str.size);
+    if(len == 0) {
+        jclass c = env->FindClass("java/lang/OutOfMemoryError");
+        env->ThrowNew(c, "Out of memory in FLValue_AsString");
+        return nullptr;
+    }
+
+    jstring retVal = toJString(env, {newBytes, len});
+    if(len != str.size) {
```

Comment:
This makes the assumption that if`newBytes` is allocated, `len` will be different from `str.size`, which isn't obvious. I think it'd be safer to initialize `newBytes` to nullptr, then you could just test for that; in fact you don't even need to because `free` allows a null parameter.